### PR TITLE
Fix completions for quote identifier given without the quote

### LINKED ISSUE #38485: [Bug]: Completions for quote identifier given without the quote
### Description

Consider the following:
image

In the above, when I click on `type`, it gets completed as `type`, whereas it should've been completed as `'type`. Same when i type `f.` and get the completion. IIRC, this used to work properly, where we used to give quoted identifiers in completions if the user had written it as a quoted identifier in the code. 

```ballerina
public function main() {
    Foo f = {<cursor1>};
    f.<cursor2>
}

type Foo record {|
    int 'type;
|};
```

### Steps to Reproduce

_No response_

### Affected Version(s)

2201.2.1

### OS, DB, other environment details and versions

_No response_

### Related area

-> Editor

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by mindula**
## Purpose
$subject

Fixes #38485

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by IMS94 — APPROVED** [review]

File: language-server/modules/langserver-core/src/test/resources/completion/expression_context/config/mapping_expr_ctx_config29.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
       "insertTextFormat": "Snippet"
     },
     {
-      "label": "version",
+      "label": "'version",
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/test/resources/completion/expression_context/config/mapping_expr_ctx_config29.json:34]: Seems like the original issue complains about the `insertText`, not the label. And I think the insertText has been fixed by another PR. Which means we don't have to do anything new to fix this.

**Review by IMS94 — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/test/resources/completion/expression_context/config/mapping_expr_ctx_config29.json (hunk 1 above)
Comment (IMS94) [reply language-server/modules/langserver-core/src/test/resources/completion/expression_context/config/mapping_expr_ctx_config29.json:34]: Just saw your comment on the issue regarding https://github.com/ballerina-platform/ballerina-lang/pull/38003. This is fine then.

**Review by IMS94 — APPROVED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java [hunk 2]
```diff
@@ -73,7 +73,7 @@ public static List<LSCompletionItem> getRecordFieldCompletionItems(BallerinaComp
             CompletionItem fieldItem = new CompletionItem();
             fieldItem.setInsertText(insertText);
             fieldItem.setInsertTextFormat(InsertTextFormat.Snippet);
-            fieldItem.setLabel(name);
+            fieldItem.setLabel(CommonUtil.escapeReservedKeyword(name));
```
Comment (pubudu91) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java:76]: From where are you getting this `name` var from? If it's from the symbol for the field, then there shouldn't be any need for you to manually escape it like this. When getting the name from the symbol it should give the correctly escaped name :thinking:

Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java:76]: We use `RecordTypeSymbol.fieldDescriptors()` map to determine the field names and their symbols. We use the keys of that map as names.

**Pull request merged**